### ISSUE TITLE
Kpt_shape not picked up

### ISSUE THREAD

**Issue by SaroshP** [ultralytics/ultralytics#18891]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello,

I am using YOLOv8 nano to custom train a model. My image annotation is as follows:
cls  xc  yc  w  h  kx  ky   visible  ----- 8 values

here is a sample annotation
0  0.089063  0.256250  0.015625  0.015625  0.089063  0.256250 1
0  0.118750  0.258594  0.021875  0.029687  0.118750  0.258594 1

I modify my yaml accordingly here to include the keypoint shape
====================================
path: 'C:/Users/Admin/Desktop/PyProjects/xDatasets/KeyPoint/'

train: 'train'
test: 'test'
val: 'val'

kpt_shape: [1,3]  # 1 keypoint, each with (x, y, visibility)

nc: 4
names: # Class Names
  0: 'uCal'
  1: 'Clips'
  2: 'Seeds'
  3: 'Areas'
================================================

but I get the following error :  **cannot reshape array of size 7 into shape (2)**

I tried to debug. I dont think it is picking up the Kpt_shape from the yaml file.

The keypoints variable in ultralytics./data/utils.py gives me NONE

I tried deleting the labels.cache file but this error continues

Thank you.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @SaroshP, thank you for reaching out and providing detailed information about your issue 🚀! We appreciate your interest in using YOLOv8 nano for custom training. 

From your description, it seems like this could potentially be a 🐛 bug related to how `kpt_shape` is being handled or parsed. To help us debug further, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/)? This would include a small dataset, the exact code used to reproduce the issue, and any relevant logs or errors.

In the meantime, here are a few suggestions:
- Ensure you have the latest version of the `ultralytics` package installed, as the issue could potentially be resolved in a newer release. To update, run:
```bash
pip install -U ultralytics
```
- Double-check both your YAML file and annotations to confirm they conform to the [YOLO format](https://docs.ultralytics.com/datasets/detect/) and the defined `kpt_shape`.

## Helpful Resources
- Please refer to our [Docs](https://docs.ultralytics.com/) for guidance on configuring keypoint models, or explore examples for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage.
- If you suspect the issue relates to how `kpt_shape` is being read, reviewing the relevant [source code](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/data/utils.py) may provide additional insight.

Once you’ve gathered the MRE or if you have additional details to share, please update this thread 👇. An Ultralytics engineer will assist you shortly!

## Join the Community
Connect with the Ultralytics community for further discussions and support:
- Real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth discussions: [Discourse](https://community.ultralytics.com/)
- Knowledge sharing: [Subreddit](https://reddit.com/r/Ultralytics)

## Verified Environments
For compatibility, we recommend running YOLO in verified environments. You can quickly try this in cloud environments with pre-configured dependencies:
- <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a><a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a><a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

## Continuous Integration
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Please let us know if you need further guidance, and we'll do our best to assist 😊!

**Comment by Y-T-G** [comment]
Can you post the full error and traceback?

**Comment by SaroshP** [comment]
> Can you post the full error and traceback?

here is the error message .. more so a warning

train: WARNING ⚠️ C:\Users\Admin\Desktop\PyProjects\xDatasets\KeyPoint\train\images\TrainingData_IQ_2D_10_tile_02-both.png: ignoring corrupt image/label: cannot reshape array of size 7 into shape (2)
train: WARNING ⚠️ C:\Users\Admin\Desktop\PyProjects\xDatasets\KeyPoint\train\images\TrainingData_IQ_2D_10_tile_02-hflip.png: ignoring corrupt image/label: cannot reshape array of size 7 into shape (2)
train: WARNING ⚠️ C:\Users\Admin\Desktop\PyProjects\xDatasets\KeyPoint\train\images\TrainingData_IQ_2D_10_tile_02-vflip.png: ignoring corrupt image/label: cannot reshape array of size 7 into shape (2)


WARNING ⚠️ No labels found in C:\Users\Admin\Desktop\PyProjects\xDatasets\KeyPoint\train\labels.cache, training may not work correctly. See https://docs.ultralytics.com/datasets for dataset formatting guidance.

It proceed to create a labels.cache with no annotations. And starts the training.

**Comment by SaroshP** [comment]
> From your description, it seems like this could potentially be a 🐛 bug related to how `kpt_shape` is being handled or parsed. To help us debug further, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/)? This would include a small dataset, the exact code used to reproduce the issue, and any relevant logs or errors.

Please see the "/minimum-reproducible-example" attached

[Sample.zip](https://github.com/user-attachments/files/18551942/Sample.zip)

**Comment by Y-T-G** [comment]
Can you post a sample of your label.txt?

**Comment by SaroshP** [comment]
sample label.txt

0  0.910937  0.256250  0.015625  0.015625  0.910937  0.256250 1
0  0.881250  0.258594  0.021875  0.029687  0.881250  0.258594 1

**Comment by SaroshP** [comment]
Label file attacted

[TrainingData_IQ_2D_2_tile_13-both.txt](https://github.com/user-attachments/files/18562635/TrainingData_IQ_2D_2_tile_13-both.txt)

**Comment by Y-T-G** [comment]
Can you try deleting labels.cache from train and val folder and running again?

**Comment by SaroshP** [comment]
I have done that multiple times ... same result

**Comment by Y-T-G** [comment]
What's your training code?

**Comment by SaroshP** [comment]
### 1. Here is the Training code
import os
import torch
from ultralytics import YOLO

device = torch.device("cuda:0") if torch.cuda.is_available() else torch.device("cpu")
torch.cuda.empty_cache()

def main():
    model = YOLO('yolov8n.yaml').load('yolov8n.pt')  # Load nano the model.
    # Resume
    # model = YOLO('C:/Users/Admin/Desktop/PyProjects/YoloV8_PIP/Demo/06_Training_Tiled_e300/weights/best.pt')
    model.info()

    train_params = {
        "project": 'Original-8n',
        "name": 'Trail-key',
        "data": 'data.yaml',

        "single_cls": False,
        "augment": True,

        "device": device,
        "imgsz": 640,
        "multi_scale": True,

        "pretrained": True,

        "epochs": 20,
        "batch": 16,  # auto batch selection (-1)

        "seed": 0,
        "cls": 0.5,
        "patience": 0,

        "optimizer": 'SGD',  # 'auto',
        "dropout": 0.1,
        "warmup_epochs": 5,  # 3
        "warmup_momentum": 0.7,  # 0.8
        "warmup_bias_lr": 0.1,  # 0.1

        "val": True,
        # "val_interval": 5,
        "save": True,
        "save_period": 10,

        "save_json": True,
        "plots": True,
        "verbose": True,

        # "lr0": 0.0001,
        # "lrf": 0.1
        "cos_lr": True,
    }

    # Training.
    results = model.train(**train_params)
    # print(results)

# save the model
# model.export("./custom_model.pt")


if __name__ == "__main__":
    main()

### 2. Here is the data.yaml 

path: 'C:/Users/Admin/Desktop/PyProjects/Sample/dataset/'

train: 'train'
test: 'valid'
val: 'valid'

nc: 4
names: # Class Names
  0: 'uCal'
  1: 'Clips'
  2: 'Seeds'
  3: 'Areas'

kpt_shape: [1,3]  # 1 keypoint, each with (x, y, visibility)

3. Attached is the Minimum Reproducible Example. So that nothing is missing. - Sample.zip

[Sample.zip](https://github.com/user-attachments/files/18566016/Sample.zip)

**Comment by Y-T-G** [comment]
`yolov8n.pt` is detection model. You should be using the pose model `yolov8n-pose.pt`

**Comment by SaroshP** [comment]
Thank you for your help !

**Comment by glenn-jocher** [comment]
@SaroshP **Response:**  
Thank you for providing the details. The error occurs because your label format should have 8 values per annotation line (class + bbox + keypoints) for `kpt_shape: [1,3]`, but your labels contain 7 values. Please ensure your labels follow the correct format: `cls x_center y_center width height kp1_x kp1_y visibility`. Also, use the **pose model** (`yolov8n-pose.pt`) instead of the detection model. For reference, see our [Pose Estimation Docs](https://docs.ultralytics.com/tasks/pose/). Let us know if this resolves the issue! 🚀
